### PR TITLE
chore(rxPageObjects): TypeDoc config & explicit types

### PR DESCRIPTION
**Pull request by jj4th**
JIRA: https://jira.rax.io/browse/FRMW-1325

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [ ] QE LGTM

During the initial rollout of the TypeDocs, several discoveries were made with regard to shortcuts and shortfalls in the initial TypeScript conversion, and especially the TypeDoc build process. The following changes are required:

1. Config changes to TypeDoc to reflect the single-file nature of the resulting library.
2. index.js should be built from TypeScript in order to maintain consistency.
3. declaration files (*.d.ts) should be generated automatically and excluded from git.
4. change watch tasks to be more TypeScript-centric
5. in order to generate the declaration files automatically, all class members should be explicitly typed.
6. rename exercise exports to avoid conflicts in documentation generation.

I chose to group these changes because all of them are necessary to support the TypeDoc config changes.

Demo:

typedoc with new navigation

### CODE REVIEW

File: utils/rx-page-objects/tsconfig.json [hunk 1]
```diff
@@ -20,7 +19,8 @@
     },
     "exclude": [],
     "include": [
-        "src/*.ts",
-        "test/*.ts"
+        "index.ts",
+        "exercise.ts",
```
Comment (parlarjb) [inline utils/rx-page-objects/tsconfig.json]: remove this

**Comment by parlarjb** [comment]
Vidyo LGTM pending change

File: utils/rx-page-objects/src/rxFieldName.exercise.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export function rxFieldName (options: IRxFieldNameExerciseOptions) {
     });
 
     return () => {
-        let component;
+        let component: component.rxFieldName;
```
Comment (CITguy) [inline utils/rx-page-objects/src/rxFieldName.exercise.ts]: Seems confusing to define `component` when we've already imported the rxFieldName page object as `component`, too.

Comment (jj4th) [inline utils/rx-page-objects/src/rxFieldName.exercise.ts]: This is how Andrew's old exercises did it.  I can raise another PR to correct this.

**Comment by jj4th** [comment]
@CITguy @parlarjb feedback addressed.

**Comment by parlarjb** [comment]
LGTM

**Comment by CITguy** [comment]
Test docs "Search" functionality isn't working, but the generated documentation is still navigable via the right hand navigation. Merging.

**Pull request merged**